README.md

```diff
@@ -99,9 +99,54 @@ These updates correspond to what React would compute using virtual-DOM diffing.
 
 When working with large immutable data structures, this approach is less performant than virtual-DOM diffing. Indeed, creating a large view out of a `Rx[List[_]]` implies that any changes to the `List` triggers a re-rendering of the entirety of the view. We plan to address this point in [#13](https://github.com/OlivierBlanvillain/monadic-html/issues/13) by combining the current approach with targeted virtual-DOM.
 
+## Interacting with js events
+
+Interactions with the imperative world of js events in done by attaching even handlers directly to xml nodes:
+
+```scala
+<button onclick={ () => println("clicked!") }>Click Me!</button>
+```
+
+Even handlers can also take one argument, in which case they will be called with raw event objects coming directly from the js world:
+
+```scala
+<div onkeydown={ (e: js.Dynamic) => () }></div>
+```
+
+The function argument can be anything here, so if you're in a type safe mood feel free to use [scala-js-dom](http://scala-js.github.io/scala-js-dom/):
+
+```scala
+<div onkeydown={ (e: dom.KeyboardEvent) => println(e.keyCode) }></div>
+```
+
+Note that the idiomatic way to handle events is to isolate side effects by opting into the `Rx` world as soon as possible. For example:
+
+```scala
+def button(text: String): (xml.Node, Rx[Unit]) = {
+  val clicked: Var[Unit] = Var(())
+  val button = <button onclick={ () => clicked := (()) }>{text}</button>
+  (button, clicked)
+}
+```
+
+## Interacting with the DOM
+
+In order to obtain references to the underlying DOM nodes, `monadic-html` implements lifecycle hooks using custom events:
+
+- `mhtml-onmount`: called when adding a node to the DOM
+- `mhtml-onunmount`: called when removing a node from the DOM
+
+In both cases, a reference to the underlying element is passed to the event hander, which can be usefull for interoperability with js libraries:
+
+```scala
+def crazyCanvasStuff(e: dom.html.Canvas): Unit = ...
+
+<canvas mhtml-onmount={ crazyCanvasStuff _ }></canvas>
```

Comment:
Nitpick. Can we write `x => crazyCanvasStuff(x)`? I think it's easier to read for some reason